src/main/java/duke/task/TaskList.java

```diff
@@ -0,0 +1,176 @@
+package duke.task;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeParseException;
+import java.util.ArrayList;
+
+import duke.exception.DukeException;
+
+/**
+ * Represents a list of tasks that needs to be done.
+ */
+public class TaskList {
+    private final ArrayList<Task> tasks;
+    private int numberOfTasks;
+
+    /**
+     * Class constructor.
+     */
+    public TaskList() {
+        tasks = new ArrayList<>();
+        numberOfTasks = 0;
+    }
+
+    /**
+     * Populates this task list with the task data stored in disk.
+     * 
+     * @param data data read from storage.
+     * @throws DukeException upon encountering incorrect format of data record.
+     */
+    public void populateWith(String[] data) throws DukeException {
+        try {
+            numberOfTasks = data.length;
+            for (int i = 0; i < numberOfTasks; i++) {
+                String tokens[] = data[i].split(",");

```

Comment:
Based on the java coding standard, the array specifiers must be attached to the type not the variable, e.g. 
```suggestion
                String[] tokens = data[i].split(",");
```

Reply:
Thanks for pointing out this coding standard violation, will resolve it in coming commits.